arelle/plugin/validate/ESEF/ESEF_Current/ValidateXbrlFinally.py

```diff
@@ -644,11 +644,30 @@ def checkFootnote(elt: ModelInlineFootnote | ModelResource, text: str) -> None:
         langsUsedByTextFacts = set()
 
         hasNoFacts = True
+        factsMissingId = []
         for qn, facts in modelXbrl.factsByQname.items():
             hasNoFacts = False
             if qn in mandatory:
                 reportedMandatory.add(qn)
             for f in facts:
+                if getDisclosureSystemYear(modelXbrl) >= 2024:
+                    if not f.id:
+                        factsMissingId.append(f)
+                    escaped = f.get("escape") in ("true", "1")
+                    if (f.concept.type.isTextBlock and not escaped) or (not f.concept.type.isTextBlock and escaped):
+                        modelXbrl.error("ESEF.2.2.7.improperApplicationOfEscapeAttribute",
+                                          _("Facts with datatype 'dtr-types:textBlockItemType' MUST use the 'escape' attribute set to 'true'. Facts with any other datatype MUST use the 'escape' attribute set to 'false' - fact %(conceptName)s"),
+                                          modelObject=f, conceptName=f.concept.qname)
+                    if f.effectiveValue == "0" and f.xValue != 0:
+                        modelXbrl.warning("ESEF.2.2.5.roundedValueBelowScaleNotNull",
+                                          _("A value that has been rounded and is below the scale should show a value of zero. It has been found to have the value %(value)s - fact %(conceptName)s"),
+                                          modelObject=f, value=f.value, conceptName=f.concept.qname)
+                    if f.concept.periodType == "instant":
+                        if f.context.instantDate.day == 1 and f.context.instantDate.month == 1:
+                            modelXbrl.error("ESEF.2.1.2.inappropriateInstantDate",
```

Comment:
Out of curiosity, would it be difficult to colocate this logic with the other logic for 2.1.2?

Reply:
It's done, the only change is that I don't check the period type based on the concept, but just if the context have a instant date.